R/calc_views.R

```diff
@@ -135,6 +137,18 @@ set_view_lim <- function(views){
       usr.lim <- views[[as.numeric(i)]][['window']][[lim.name]][1:2]
       views[[as.numeric(i)]][['window']][[lim.name]] <- data.lim
       views[[as.numeric(i)]][['window']][[lim.name]][!is.na(usr.lim)] <- usr.lim[!is.na(usr.lim)]
+      
+      usr.axs <- data[[paste0(var, 'axs')]][[as.numeric(i)]]
+      view.index <- which(!names(views[[as.numeric(i)]]) %in% c('window', 'gsplot'))[which(!is.na(usr.axs))]
+      
+      if (any(!is.na(usr.axs)) && usr.axs[which(!is.na(usr.axs))] == 'o') {
+        buffer <- 0.04*views[[as.numeric(i)]][['window']][[lim.name]][[2]] #4% buffer based on upper limit
+        buffer.lim <- views[[as.numeric(i)]][['window']][[lim.name]][[which(is.na(usr.lim))]]
+        buffer.lim <- ifelse(which(is.na(usr.lim)) == 1, buffer.lim - buffer, buffer.lim + buffer)
+        views[[as.numeric(i)]][['window']][[lim.name]][[which(is.na(usr.lim))]] <- buffer.lim
+        views[[as.numeric(i)]][[view.index]][[paste0(var, 'axs')]] <- NULL
+        views[['par']][[paste0(var, 'axs')]] <- 'i'
```

Comment:
The `default.yaml` has `yaxs = 'r'` and `xaxs = 'r'`, so when the "o" style is selected we add the 4% buffer to one side and so will want the axes to have the 'i' style so that they have the exact limits we defined.